### PR TITLE
[DS-119]: feat: adds bulkActions showcase

### PR DESCRIPTION
**Pull request by michalisKout**
## Description

[jira task](https://orfium.atlassian.net/secure/RapidBoard.jspa?rapidView=40&modal=detail&selectedIssue=DS-119)

- adds bulk actions showcase because this will help other devs to understand the logic of the bulk actions selector.
- adds button style configuration prop for SectionHeader component
- fixes state update issues due to complex logic on multi-selection

## Screenshot

showcase can be previewed [here](https://ictinus-feat-asset-matc-fu99r7.herokuapp.com/?path=/story/design-system-asset-matching--bulk-actions-showcase)

## Test Plan

updates snapshots

### CODE REVIEW

**Comment by michalisKout** [comment]
@panvourtsis should we add this as new story to Jira's DS board?

**Comment by michalisKout** [comment]
**UPDATE** : i am adding some style config for bulk actions header

**Comment by panvourtsis** [comment]
@michalisKout yes i dont understand by the description what i am about to see, the ticket would help for this + tracking

**Comment by michalisKout** [comment]
> @michalisKout yes i dont understand by the description what i am about to see, the ticket would help for this + tracking

Ok! I updated the description with the links for the [Jira task](https://orfium.atlassian.net/secure/RapidBoard.jspa?rapidView=40&modal=detail&selectedIssue=DS-119) and the [playground](https://ictinus-feat-asset-matc-fu99r7.herokuapp.com/?path=/story/design-system-asset-matching--bulk-actions-showcase) for the bulk actions section.

**Review by kostasdano — APPROVED** [review]

File: src/components/AssetMatching/components/CheckBoxContainer/CheckBoxContainer.tsx [hunk 1]
```diff
@@ -10,19 +10,21 @@ interface Props {
   score?: string | number;
   isEnabled: boolean;
   isChecked: boolean;
-  customContent?: JSX.Element;
+  customCheckboxContent?: JSX.Element | null;
+  intermediateStatus?: boolean;
```
Comment (panvourtsis) [inline src/components/AssetMatching/components/CheckBoxContainer/CheckBoxContainer.tsx:14]: `isIntermediateStatus`

File: src/components/AssetMatching/components/SectionHeader/SectionHeader.tsx [hunk 2]
```diff
@@ -14,34 +14,43 @@ interface Props {
...
-  isButtonFilled?: boolean;
-  customCheckboxContent?: JSX.Element;
+  buttonStyles?: {
+    secondaryButtonColor?: string;
+    primaryButtonColor?: string;
+    isButtonFilled?: boolean;
+  };
+  customCheckboxContent?: JSX.Element | null;
   isChecked?: boolean;
+  intermediateStatus?: boolean;
```
Comment (panvourtsis) [inline src/components/AssetMatching/components/SectionHeader/SectionHeader.tsx:24]: same here

**Review by panvourtsis — APPROVED** [review]

**Pull request merged**